gems/pending/util/thumb_print.rb

```diff
@@ -0,0 +1,29 @@
+require 'net/http'
+require 'openssl'
+
+class ThumbPrint
+  attr_reader :thumb_print, :der_thumb_print, :x509_cert, :host
+
+  def initialize(host)
+    $log.info "ThumbPrint.initialize(#{host})" if $log
+    @host = host
+    uri
```

Comment:
Yeah, I think this feels weird to call it here and then not use the return value.  As I said earlier, writing it like an accessor method and then never using @uri allows it to be built when needed automatically.